src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -147,72 +148,163 @@ handle_t halide_hexagon_remote_get_symbol(handle_t module_ptr, const char* name,
 
 volatile int power_ref_count = 0;
 
-int halide_hexagon_remote_power_hvx_on() {
-    if (power_ref_count == 0) {
-        HAP_power_response_t power_info;
+int halide_hexagon_HAP_power_hvx_on(halide_hvx_power_perf_t *perf) {
```

Comment:
I understand, but as long as you are forced to have that function exist, you might as well just make that the "base" level functionality. It eliminates a lot of code, and I think it's safer to only have one function that increments the power_ref_count.